src/Agent.Listener/Configuration/AutoLogonManager.cs

```diff
@@ -0,0 +1,198 @@
+#if OS_WINDOWS
+using Microsoft.VisualStudio.Services.Agent;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Security.Principal;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+    [ServiceLocator(Default = typeof(AutoLogonManager))]
+    public interface IAutoLogonManager : IAgentService
+    {
+        Task ConfigureAsync(CommandSettings command);
+        void Unconfigure();
+    }
+
+    public class AutoLogonManager : AgentService, IAutoLogonManager
+    {
+        private ITerminal _terminal;
+        private INativeWindowsServiceHelper _windowsServiceHelper;
+        private IAutoLogonRegistryManager _autoLogonRegManager;
+        private IConfigurationStore _store;
+
+        public override void Initialize(IHostContext hostContext)
+        {
+            base.Initialize(hostContext);
+            _terminal = hostContext.GetService<ITerminal>();
+            _windowsServiceHelper = hostContext.GetService<INativeWindowsServiceHelper>();
+            _autoLogonRegManager = HostContext.GetService<IAutoLogonRegistryManager>();
+            _store = hostContext.GetService<IConfigurationStore>();
+        }
+
+        public async Task ConfigureAsync(CommandSettings command)
+        {
+            if (!_windowsServiceHelper.IsRunningInElevatedMode())
+            {
+                Trace.Error("Needs Administrator privileges to configure agent with AutoLogon capability.");
+                Trace.Info("You will need to unconfigure the agent and then re-configure with Administrative rights");
```

Comment:
this is duplicate with the Exception string, remove this.
`Trace.Info("You will need to unconfigure the agent and then re-configure with Administrative rights");`